File: problem-set-6/sentimental-mario-more/mario.py
def print_hashes(i):
    # Print i+1 hashes (to match the 0-based indexing in the C code)
    print("#" * (i + 1), end="")

def main():
    # Get the height input from the user
    while True:
        try:
            n = int(input("Height: "))
            if 1 <= n <= 8:
                break
            else:
                print("Please enter a number between 1 and 8.")
        except ValueError:
            print("Please enter a valid integer.")

    # Print the pyramid pattern
    for i in range(n):
        # Print the leading spaces
        print(" " * (n - i - 1), end="")

        # Print left hashes
        print_hashes(i)

        # Print the gap between the two sides of the pyramid
        print("  ", end="")

        # Print right hashes
        print_hashes(i)

        # Move to the next line
        print()

# Run the main function
if __name__ == "__main__":
    main()

File: problem-set-1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_hashes(int i);

int main(void)
{
    int n;

    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < (n - i) - 1; j++)
        {
            printf(" ");
        }

        print_hashes(i);

        printf("  ");

        print_hashes(i);

        printf("\n");
    }
}

void print_hashes(int i)
{
    for (int l = 0; l <= i; l++)
    {
        printf("#");
    }
}
